packages/sshnoports/lib/sshnp/sshnp.dart

```diff
@@ -146,19 +152,27 @@ abstract class SSHNP {
 
   abstract final bool direct;
 
+  /// If ssh tunnel is unused (no active connections via port forwards) for
+  /// longer than this many seconds, then the connection will be closed.
+  /// Defaults to [defaults.defaultIdleTimeout]
+  abstract int idleTimeout;
+
+  /// The ssh client to use when doing outbound ssh within this program
+  abstract SupportedSshClient sshClient;
+
+  /// Completes when the SSHNP instance is no longer doing anything
+  /// e.g. controlling a direct ssh tunnel using the pure-dart SSHClient
+  Future<void> get done;
+
   /// Default parameters for sshnp
   static const defaultDevice = 'default';
   static const defaultPort = 22;
   static const defaultLocalPort = 0;
   static const defaultSendSshPublicKey = '';
   static const defaultLocalSshOptions = <String>[];
-  static const defaultVerbose = false;
-  static const defaultRsa = false;
-  static const defaultRootDomain = 'root.atsign.org';
-  static const defaultSshrvGenerator = SSHRV.localBinary;
-  static const defaultLocalSshdPort = 22;
   static const defaultLegacyDaemon = true;
   static const defaultListDevices = false;
+  static const defaultSshClient = SupportedSshClient.hostSsh;
```

Comment:
Wouldn't it be best to move the rest of these to `common/defaults.dart`

Reply:
Not a blocker, but can be done in a later PR

Reply:
I didn't move these few as I felt like they are specific to SSHNP - legacyDaemon and listDevices for sure, and we may in future have different default sshClients in daemon / client

Reply:
> I didn't move these few as I felt like they are specific to SSHNP - legacyDaemon and listDevices for sure, and we may in future have different default sshClients in daemon / client

That makes sense